lib/UserBackend.php

```diff
@@ -160,6 +162,34 @@ public function setPassword(string $uid, string $password): bool {
 		return false;
 	}
 
+	public function getPasswordHash(string $userId): ?string {
+		if (!$this->userExists($userId)) {
+			return null;
+		}
+		$qb = $this->dbConn->getQueryBuilder();
+		$qb->select('password')
+			->from('guests_users')
+			->where($qb->expr()->eq('uid_lower', $qb->createNamedParameter(mb_strtolower($userId))));
+		/** @var false|string $hash */
+		$hash = $qb->executeQuery()->fetchOne();
+		if ($hash === false) {
+			return null;
+		}
+		return $hash;
+	}
+
+	public function setPasswordHash(string $userId, string $passwordHash): bool {
+		$qb = $this->dbConn->getQueryBuilder();
+		$qb->update('guests_users')
+			->set('password', $qb->createNamedParameter($passwordHash))
+			->where($qb->expr()->eq('uid_lower', $qb->createNamedParameter(mb_strtolower($userId))));
+		$result = $qb->executeStatement();
+		if ($result === 0) {
+			return false;
+		}
+		return true;
```

Comment:
```suggestion
		return ($result !== 0);
```
Also would be nice to check the hash string is in a format we expect before setting in DB.

Reply:
Needs https://github.com/nextcloud/server/pull/46186